editor/src/components/canvas/ui-jsx-canvas.tsx

```diff
@@ -331,7 +334,7 @@ export const UiJsxCanvas = betterReactMemo(
         jsxFactoryFunctionName: jsxFactoryFunction,
       })
 
-      const topLevelElementsMap = new Map(topLevelJsxComponents)
+      const topLevelElementsMap = useKeepReferenceEqualityIfPossible(new Map(topLevelJsxComponents))
```

Comment:
this was the one liner to keep the canvas root from rerendering. However, I don't remember why was `topLevelJsxComponents` here copied into a `new Map()`? Is this a feature we still rely on? I'm too sleepy to figure out but maybe @Rheeseyb remembers something...

Reply:
https://github.com/concrete-utopia/dystopia/pull/3457/files#r595547751